samcli/commands/init/init_templates.py

```diff
@@ -262,3 +265,26 @@ def template_does_not_meet_filter_criteria(
         or (package_type and package_type != template.get("packageType"))
         or (dependency_manager and dependency_manager != template.get("dependencyManager"))
     )
+
+
+def filter_value_matches_template_runtime(filter_value, template_runtime):
+    """
+    Validate if the filter value matches template runtimes from the manifest file
+
+    Parameters
+    ----------
+    filter_value : str
+        Lambda runtime used to filter through data generated from the manifest
+    template_runtime : str
+        Runtime of the template in view
+
+    Returns
+    -------
+    bool
+        True if there is a match else False
+    """
+    if is_custom_runtime(filter_value) and filter_value != get_provided_runtime_from_custom_runtime(template_runtime):
```

Comment:
If `filter_value` is `None` there is no need to run rest of this function. So we can simplify this by adding;
```
if not filter_value:
  return True
```
at the beginning of the function.